From 6f4e66f87ccf10d8dff6b764ece0ad33b2191b63 Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Tue, 3 Jan 2012 14:59:08 +0100
Subject: [PATCH] Drop obsolete g_thread_init()

Since version 2.2 g_type_init() initializes also the thread
system so g_thread_init() is not required anymore. It is deprecated since glib
2.31.

Bump glib requirement accordingly.

https://bugzilla.gnome.org/show_bug.cgi?id=650061
---
 configure.ac                      |    2 +-
 examples/continuous-twitter.c     |    1 -
 examples/get-fireeagle-location.c |    1 -
 examples/get-flickr-favorites.c   |    1 -
 examples/lastfm-shout.c           |    1 -
 examples/post-twitter.c           |    1 -
 examples/test-raw.c               |    1 -
 examples/test-xml.c               |    1 -
 tests/custom-serialize.c          |    1 -
 tests/flickr.c                    |    1 -
 tests/lastfm.c                    |    1 -
 tests/oauth-async.c               |    1 -
 tests/oauth.c                     |    1 -
 tests/oauth2.c                    |    1 -
 tests/proxy-continuous.c          |    1 -
 tests/proxy.c                     |    1 -
 tests/threaded.c                  |    1 -
 tests/xml.c                       |    1 -
 18 files changed, 1 insertions(+), 18 deletions(-)

Index: librest-0.7.12/examples/continuous-twitter.c
===================================================================
--- librest-0.7.12.orig/examples/continuous-twitter.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/examples/continuous-twitter.c	2012-05-27 21:38:16.607431425 +0800
@@ -43,7 +43,6 @@
   char pin[256];
   GMainLoop *loop;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   loop = g_main_loop_new (NULL, FALSE);
Index: librest-0.7.12/examples/get-fireeagle-location.c
===================================================================
--- librest-0.7.12.orig/examples/get-fireeagle-location.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/examples/get-fireeagle-location.c	2012-05-27 21:38:16.607431425 +0800
@@ -34,7 +34,6 @@
   RestXmlParser *parser;
   RestXmlNode *root, *node;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   /* Create the proxy */
Index: librest-0.7.12/examples/get-flickr-favorites.c
===================================================================
--- librest-0.7.12.orig/examples/get-flickr-favorites.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/examples/get-flickr-favorites.c	2012-05-27 21:38:16.611431356 +0800
@@ -106,7 +106,6 @@
   char *frob, *url;
   const char *token;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   proxy = flickr_proxy_new ("cf4e02fc57240a9b07346ad26e291080", "cdfa2329cb206e50");
Index: librest-0.7.12/examples/lastfm-shout.c
===================================================================
--- librest-0.7.12.orig/examples/lastfm-shout.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/examples/lastfm-shout.c	2012-05-27 21:38:16.611431356 +0800
@@ -87,7 +87,6 @@
     { NULL }
   };
 
-  g_thread_init (NULL);
   g_type_init ();
 
   context = g_option_context_new ("- send a shout to a Last.fm user");
Index: librest-0.7.12/examples/post-twitter.c
===================================================================
--- librest-0.7.12.orig/examples/post-twitter.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/examples/post-twitter.c	2012-05-27 21:38:16.611431356 +0800
@@ -31,7 +31,6 @@
   GError *error = NULL;
   char pin[256];
 
-  g_thread_init (NULL);
   g_type_init ();
 
   if (argc != 2) {
Index: librest-0.7.12/examples/test-raw.c
===================================================================
--- librest-0.7.12.orig/examples/test-raw.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/examples/test-raw.c	2012-05-27 21:38:16.611431356 +0800
@@ -48,7 +48,6 @@
   gssize len;
 
   g_type_init ();
-  g_thread_init (NULL);
 
   loop = g_main_loop_new (NULL, FALSE);
 
Index: librest-0.7.12/examples/test-xml.c
===================================================================
--- librest-0.7.12.orig/examples/test-xml.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/examples/test-xml.c	2012-05-27 21:38:16.611431356 +0800
@@ -114,7 +114,6 @@
   GMainLoop *loop;
 
   g_type_init ();
-  g_thread_init (NULL);
 
   loop = g_main_loop_new (NULL, FALSE);
 
Index: librest-0.7.12/tests/custom-serialize.c
===================================================================
--- librest-0.7.12.orig/tests/custom-serialize.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/custom-serialize.c	2012-05-27 21:38:16.611431356 +0800
@@ -124,7 +124,6 @@
   int i;
   GError *error;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   session = soup_session_sync_new ();
Index: librest-0.7.12/tests/flickr.c
===================================================================
--- librest-0.7.12.orig/tests/flickr.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/flickr.c	2012-05-27 21:38:16.611431356 +0800
@@ -36,7 +36,6 @@
   RestXmlParser *parser;
   RestXmlNode *root, *node;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   /* Create the proxy */
Index: librest-0.7.12/tests/lastfm.c
===================================================================
--- librest-0.7.12.orig/tests/lastfm.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/lastfm.c	2012-05-27 21:38:16.611431356 +0800
@@ -37,7 +37,6 @@
   RestXmlParser *parser;
   RestXmlNode *root, *u_node, *node;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   /* Create the proxy */
Index: librest-0.7.12/tests/oauth-async.c
===================================================================
--- librest-0.7.12.orig/tests/oauth-async.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/oauth-async.c	2012-05-27 21:38:16.611431356 +0800
@@ -113,7 +113,6 @@
   OAuthProxy *oproxy;
   GError *error = NULL;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   /* Install a timeout so that we don't hang or infinite loop */
Index: librest-0.7.12/tests/oauth.c
===================================================================
--- librest-0.7.12.orig/tests/oauth.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/oauth.c	2012-05-27 21:38:16.611431356 +0800
@@ -33,7 +33,6 @@
   RestProxyCall *call;
   GError *error = NULL;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   /* Create the proxy */
Index: librest-0.7.12/tests/oauth2.c
===================================================================
--- librest-0.7.12.orig/tests/oauth2.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/oauth2.c	2012-05-27 21:38:16.611431356 +0800
@@ -47,7 +47,6 @@
 int
 main (int argc, char **argv)
 {
-  g_thread_init (NULL);
   g_type_init ();
 
   test_extract_token ();
Index: librest-0.7.12/tests/proxy-continuous.c
===================================================================
--- librest-0.7.12.orig/tests/proxy-continuous.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/proxy-continuous.c	2012-05-27 21:38:16.611431356 +0800
@@ -168,7 +168,6 @@
   char *url;
   RestProxy *proxy;
 
-  g_thread_init (NULL);
   g_type_init ();
   loop = g_main_loop_new (NULL, FALSE);
 
Index: librest-0.7.12/tests/proxy.c
===================================================================
--- librest-0.7.12.orig/tests/proxy.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/proxy.c	2012-05-27 21:38:16.611431356 +0800
@@ -283,7 +283,6 @@
   char *url;
   RestProxy *proxy;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   session = soup_session_async_new ();
Index: librest-0.7.12/tests/threaded.c
===================================================================
--- librest-0.7.12.orig/tests/threaded.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/threaded.c	2012-05-27 21:38:16.611431356 +0800
@@ -85,7 +85,6 @@
   char *url;
   int i;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   session = soup_session_sync_new ();
Index: librest-0.7.12/tests/xml.c
===================================================================
--- librest-0.7.12.orig/tests/xml.c	2012-05-27 21:38:06.167613589 +0800
+++ librest-0.7.12/tests/xml.c	2012-05-27 21:38:16.611431356 +0800
@@ -33,7 +33,6 @@
   RestXmlNode *root, *node;
   char *xml;
 
-  g_thread_init (NULL);
   g_type_init ();
 
   parser = rest_xml_parser_new ();
